idunn/api/directions.py

```diff
@@ -15,25 +20,35 @@
 
 
 def get_directions(
+    response: Response,
     f_lon: float,
     f_lat: float,
     t_lon: float,
     t_lat: float,  # URL values
     request: Request,
-    type: str = "",
+    type: constr(min_length=1),
     language: str = "en",  # query parameters
 ):
     rate_limiter.check_limit_per_client(request)
 
     from_position = (f_lon, f_lat)
     to_position = (t_lon, t_lat)
 
-    if not type:
-        raise HTTPException(status_code=400, detail='"type" query param is required')
+    if type in ("publictransport", "taxi", "vtc", "carpool"):
+        allowed_zone = any(
```

Comment:
Maybe we can put this logic in a separate function, even in the `geometry.py` file itself, so it can be maintained and tested separately.

Reply:
I totally agree, just moved it here: https://github.com/QwantResearch/idunn/pull/101/files#diff-11182b086324030b6623bf1b6cdcce50R40-R51